src/imagestyle/converters.js

```diff
@@ -49,41 +47,27 @@ export function viewToModelStyleAttribute( styles ) {
 	const filteredStyles = styles.filter( style => !style.isDefault );
 
 	return ( evt, data, consumable, conversionApi ) => {
-		for ( const style of filteredStyles ) {
-			viewToModelImageStyle( style, data, consumable, conversionApi );
+		if ( !data.output ) {
+			return;
 		}
-	};
-}
-
-// Converter from view to model converting single style.
-// For more information see {@link module:engine/conversion/viewconversiondispatcher~ViewConversionDispatcher};
-//
-// @param {module:image/imagestyle/imagestyleengine~ImageStyleFormat} style
-// @param {Object} data
-// @param {module:engine/conversion/viewconsumable~ViewConsumable} consumable
-// @param {Object} conversionApi
-function viewToModelImageStyle( style, data, consumable, conversionApi ) {
-	const viewFigureElement = data.input;
-	const modelImageElement = data.output;
-
-	// *** Step 1: Validate conversion.
-	// Check if view element has proper class to consume.
-	if ( !consumable.test( viewFigureElement, { class: style.className } ) ) {
-		return;
-	}
 
-	// Check if figure is converted to image.
-	if ( !isImage( modelImageElement ) ) {
-		return;
-	}
+		const viewFigureElement = data.input;
+		const modelImageElement = data.output.getItems().next().value;
```

Comment:
Since `writer.setAttribute` already accepts range, we should make `schema.checkAttribute` also accepts range and we will be able to get rid off this ugly line. In fact, it is schema context what should accept range. It should build the context based on the first element in that range or throw an exception if there are more elements.

Reply:
Or you can use `first( output.getItems() );` ;)

Reply:
For now I'll change it to `first()`. I'm already using `first()` in other image converters to get range first item.